### ISSUE TITLE
Windows CLI suite still leaks analysis warnings into ConsoleCapture

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#4720]
## Summary

The Windows CLI suite can still route an ambient performance-analysis warning
into an unrelated `ConsoleCapture` assertion after #4324 closed #3446.

## Observed

PR #4616 run
[32996775184](https://github.com/richlander/dotnet-inspect/actions/runs/32996775184)
failed `test-windows` on exact head
`79bac625620e7a1382801300a1de6ea33b7d8136`.

Four tests failed only at `Assert.Empty(error)`:

- `PerformanceTriageOrderByCallerLoopDepth_PutsMissingEvidenceLast("asc")`
- `PerformanceTriageOrderByCallerLoopDepth_PutsMissingEvidenceLast("desc")`
- `PerformanceTriage_ExposesDirectCallerLoopEvidenceWithoutChangingLocalLoop("library")`
- `PerformanceTriageAllocationFanout_IncludesDirectCallerLoopEvidence`

Each captured:

```text
Warning: performance analysis incomplete for
DotnetInspector.Tests.SampleStateMachineAsyncClass::AttributedStateMachineAsync:
BadImageFormatException: The classic async source does not map to a unique
valid state-machine body.
```

The same warning appeared in the Windows log for independently green run
[32996640388](https://github.com/richlander/dotnet-inspect/actions/runs/32996640388),
which passed all 4,748 CLI tests. The failing and green runs overlapped in time,
used the same Windows suite, and both emitted the warning before the affected
test region. This makes the branch's scheduling/order the differentiator, not
the warning-producing analysis behavior.

## Expected

Warnings produced while another command or test analyzes the test assembly
must not enter a capture owned by a command inspecting
`ILInspector.Analysis.Fixtures.dll`.

## Notes

#4324 placed direct `ConsoleCapture` owners in an assembly-exclusive collection.
This recurrence suggests either a delayed write that outlives its owning
capture or another shared diagnostic/output path not covered by that
isolation. Do not weaken the empty-stderr assertions; identify and isolate the
writer or lifetime leak.

**Comment by richlander** [comment]
Correction: this was not a ConsoleCapture recurrence. PR #4616 had placed deliberately malformed ownership specimens in the shared ILInspector.Analysis.Fixtures.dll, so CallerLoop CLI tests correctly captured those diagnostics. The specimens are being moved to a dedicated fixture assembly on #4616; the six affected parameterized cases and the malformed-ownership gate pass after isolation.
